contracts/multisig-prover/src/encoding/stacks/execute_data.rs

```diff
@@ -0,0 +1,273 @@
+use axelar_wasm_std::hash::Hash;
+use cosmwasm_std::HexBinary;
+use error_stack::ResultExt;
+use k256::ecdsa::RecoveryId;
+use multisig::key::Signature;
+use multisig::msg::SignerWithSig;
+use multisig::verifier_set::VerifierSet;
+use stacks_clarity::common::codec::StacksMessageCodec;
+use stacks_clarity::common::types::StacksEpochId;
+use stacks_clarity::vm::representations::ClarityName;
+use stacks_clarity::vm::types::signatures::{
+    BufferLength, ListTypeData, SequenceSubtype, TypeSignature,
+};
+use stacks_clarity::vm::types::{TupleData, Value};
+
+use crate::encoding::stacks::{ecdsa_key, encode_messages, payload_digest, WeightedSigners};
+use crate::error::ContractError;
+use crate::payload::Payload;
+
+pub const APPROVE_MESSAGES_FUNCTION: &str = "approve-messages";
+pub const ROTATE_SIGNERS_FUNCTION: &str = "rotate-signers";
+
+pub struct Proof {
+    pub signers: WeightedSigners,
+    pub signatures: Vec<Value>,
+}
+
+impl Proof {
+    pub fn new(
+        verifier_set: &VerifierSet,
+        signers_with_sigs: Vec<SignerWithSig>,
+    ) -> Result<Self, ContractError> {
+        let signers = WeightedSigners::try_from(verifier_set)?;
+
+        let mut signers_with_sigs: Vec<(Vec<u8>, Signature)> = signers_with_sigs
+            .into_iter()
+            .map(|signer| {
+                let key = ecdsa_key(&signer.signer.pub_key).expect("not ecdsa key");
```

Comment:
can we return an error here, and not expect?

Reply:
Done